File: VB/MapControl_SearchPanel/Form1.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraMap

Namespace MapControl_SearchPanel

    Public Partial Class Form1
        Inherits Form

        Private ReadOnly Property SearchLayer As InformationLayer
            Get
                Return CType(mapControl1.Layers("SearchLayer"), InformationLayer)
            End Get
        End Property

        Private ReadOnly Property SearchProvider As BingSearchDataProvider
            Get
                Return CType(SearchLayer.DataProvider, BingSearchDataProvider)
            End Get
        End Property

        Public Sub New()
            InitializeComponent()
            AddHandler SearchLayer.DataRequestCompleted, AddressOf SearchLayer_DataRequestCompleted
            ' 
            mapControl1.Layers.Add(New ImageLayer() With {.DataProvider = New BingMapDataProvider() With {.BingKey = "YOUR BING KEY", .Kind = BingMapKind.Area}})
        End Sub

        Private Sub SearchLayer_DataRequestCompleted(ByVal sender As Object, ByVal e As RequestCompletedEventArgs)
            mapControl1.ZoomToFitLayerItems()
        End Sub

        Private Sub bSearch_Click(ByVal sender As Object, ByVal e As EventArgs)
            SearchProvider.Search(teKeyword.Text)
        End Sub
    End Class
End Namespace

File: CS/MapControl_SearchPanel/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraMap;

namespace MapControl_SearchPanel {
    public partial class Form1 : Form {
        InformationLayer SearchLayer { get { return (InformationLayer)mapControl1.Layers["SearchLayer"]; } }
        BingSearchDataProvider SearchProvider { get { return (BingSearchDataProvider)SearchLayer.DataProvider; } }

        public Form1() {
            InitializeComponent();
            SearchLayer.DataRequestCompleted += SearchLayer_DataRequestCompleted;
            // 
            mapControl1.Layers.Add(new ImageLayer() {
                DataProvider = new BingMapDataProvider() {
                    BingKey = "YOUR BING KEY",
                    Kind = BingMapKind.Area
                }
            });
        }

        void SearchLayer_DataRequestCompleted(object sender, RequestCompletedEventArgs e) {
            mapControl1.ZoomToFitLayerItems();
        }

        private void bSearch_Click(object sender, EventArgs e) {
            SearchProvider.Search(teKeyword.Text);
        }
    }
}
